src/utils.js

```diff
@@ -520,7 +520,29 @@ export class DDGProxy {
     }
 }
 
+const maxCounter = new Map()
+function numberOfTimesDebugged (feature, message) {
+    let key
+    try {
+        // Could throw if message is not serializable
+        key = JSON.stringify({ feature, message })
+    } catch {
+        return 0
+    }
+    if (!maxCounter.has(key)) {
+        maxCounter.set(key, 1)
+    } else {
+        maxCounter.set(key, maxCounter.get(key) + 1)
+    }
+    return maxCounter.get(key)
+}
+
+const DEBUG_MAX_TIMES = 1000
+
 export function postDebugMessage (feature, message) {
+    if (numberOfTimesDebugged(feature, message) > DEBUG_MAX_TIMES) {
```

Comment:
I guess rate-limiting instead of a max_times would not be helpful for debugging?

Reply:
As in just preventing _n_ messages over _y_ window of time from happening?

I think that would be less clear unless we provided info about the truncation, the alternative is telling the debugger we've rate limited but that involves changes on the extension too, on page load we get a lot of messages sometimes so that threshold initially might be very high.

Ultimately this is just a quick fix to prevent it breaking the debugger we could always revisit it.